Cython/Utility/CppConvert.pyx

```diff
@@ -235,3 +235,24 @@ cdef object {{cname}}(const std_complex[X]& z):
     tmp.real = <double>z.real()
     tmp.imag = <double>z.imag()
     return tmp
+
+#################### enum_class.from_py ####################
```

Comment:
This stuff looks simple enough that I wonder if it really needs utility code - why not just generate the C++ casts directly?

Reply:
It's possible that the converted utility code is actually doing more type-checking etc than I'm giving it credit for, of course...

Reply:
Apologies - what does "generate the C++ casts directly" mean? We basically need to be able to convert Python objects to and from `underlying_type` (which is only known at runtime). What's the best way to do that?

Reply:
Looking at it again, it doesn't seem completely unreasonable - it obviously has to generate things like overflow-checks so the code is doing more than it initially looks.

However - here's what I'd try:

The conversions go `object -> int_type -> enum` and `enum -> int_type -> object`.
The `object <-> int_type` is taken from `CIntLike` so use their implementation. `create_from_py_utility_code` can become something like:

    def create_from_py_utility_code(self, env):
       if self.from_py_function:
            return True
       if CIntLike.create_from_py_utility_code(self, env):
           self.from_py_function = 'static_cast<%s>(%s)' % (
                self.cname, self.from_py_function)
       return False

i.e. all you've really done is wrapped the existing conversion in a cast.

It's possible I've missed something important here and it won't work, but I think it's worth a go.

Reply:
Yes! Something like this works, except that for `to_py_function`, I needed to get clever and use C++ lambdas, to make it a callable that correctly does the two chained casts:

```python
    def create_to_py_utility_code(self, env):                                                                                                                                     
        if self.to_py_function is not None:                                                                                                                                       
            return True                                                                                                                                                           
        context = {}                                                                                                                                                              
        if self.from_py_function:                                                                                                                                                 
            return True                                                                                                                                                           
        if self.underlying_type.create_to_py_utility_code(env):                                                                                                                   
            self.to_py_function = '[](const %s& x){return %s((%s)x);}' % (                                                                                                        
                self.cname,                                                                                                                                                       
                self.underlying_type.to_py_function,                                                                                                                              
                self.underlying_type.empty_declaration_code()                                                                                                                     
            )                                                                                                                                                                     
        return True 
```